components/mosart/src/riverroute/MOSART_sediment_mod.F90

```diff
@@ -0,0 +1,492 @@
+    use shr_sys_mod   , only : shr_sys_abort
+    !use clm_varcon , only : denh2o, grav !!density of liquid water [kg/m3], gravity constant [m/s2]
+    use RunoffMod, only : Tctl, TUnit, TRunoff, TPara
+    use RunoffMod, only : rtmCTL
+    use rof_cpl_indices, only : nt_rtm, rtm_tracers, nt_nliq, nt_nice, nt_nmud, nt_nsan
+    use MOSART_BGC_type, only : TSedi, TSedi_para
+    implicit none
+    real(r8), parameter :: TINYVALUE_s = 1.0e-12_r8  ! double precision variable has a significance of about 16 decimal digits
+    real(r8), parameter :: densedi = 2650._r8      ! density of sediment [Kg/m3]
+    real(r8), parameter :: diam_sand = 0.00035_r8  ! grain size of sand, 0.35mm
+    real(r8), parameter :: KVIS = 1.0036e-6_r8     ! the kinematic viscosity (m2/s)at 20°Ê*/
+    real(r8), parameter :: DVIS = 1.002e-3_r8      ! the dynamatic viscosity (Pa°§s)at 20°Ê*/
+
+! !PUBLIC MEMBER FUNCTIONS:
+    contains
+
+    subroutine hillslopeSediment(iunit, theDeltaT)
+    ! !DESCRIPTION: Hillslope erosion/routing of sediment
+        implicit none
+        
+        integer, intent(in) :: iunit
+        real(r8), intent(in) :: theDeltaT
+
+        real(r8) :: Tau_h
+        TRunoff%ehout(iunit, nt_nmud) = -TRunoff%qsur(iunit,nt_nmud) !0._r8
+        TRunoff%qsub(iunit, nt_nmud) = 0._r8
+        !TRunoff%etin(iunit,nt_nmud) = 0._r8
+        !TRunoff%etin(iunit,nt_nmud) = TRunoff%qsur(iunit,nt_nmud) * TUnit%area(iunit)
+        
+        !TRunoff%etin(iunit,nt_nmud) = 1.e-6_r8 * TUnit%area(iunit) * TUnit%frac(iunit) ! Ks/s
+        !TRunoff%etin(iunit,nt_nsan) = 1.e-6_r8 * TUnit%area(iunit) * TUnit%frac(iunit) ! Ks/s
+    end subroutine hillslopeSediment
+
+    subroutine subnetworkSediment(iunit, theDeltaT)
+    ! !DESCRIPTION: subnetwork channel processes for mud sediment. Assume no sand sediment in tributary channels
+        implicit none    
+        integer, intent(in) :: iunit !, nt      
+        real(r8), intent(in) :: theDeltaT
+
+        !real(r8) :: ers       ! sand-sediment erosion
+        !real(r8) :: ersal     ! sand-sediment erosion @ active layer
+        !real(r8) :: ersb      ! sand-sediment erosion @ bank
+        !real(r8) :: ses       ! sand-sediment settled to bed zone
+        !real(r8) :: seout     ! sand-sediment discharge out of local channel
+        real(r8) :: erm       ! mud-sediment erosion
+        real(r8) :: ermal     ! mud-sediment erosion @ active layer
+        real(r8) :: ermb      ! mud-sediment erosion @ bank
+        real(r8) :: sem       ! mud-sediment settled to bed zone
+        real(r8) :: meout     ! mud-sediment discharge out of local channel
+        real(r8) :: erb       ! total-sediment erosion @ bank
+        real(r8) :: temp1, temp2, w_temp
+
+
+        real(r8) :: Achannel  ! effective channel area
+        
+            ! sand-sediment
+            !ses = CRSES(iunit,TRunoff%yt(iunit,nt_nliq),TUnit%tslp(iunit),TRunoff%conc_t(iunit,nt_nsan))
```

Comment:
Delete commented out lines

Reply:
Not sure if @liho745 wants to keep this..

Reply:
yes, we can delete them.